### DISCUSSION TITLE
Some clarity needed - Why does "Open" try to run a file Sandboxed instead of using Host's default program?

### DISCUSSION THREAD

**Discussion by TFWol** [sandboxie-plus/Sandboxie#3589] [Q&A Feedback] [answered]
![image](https://github.com/sandboxie-plus/Sandboxie/assets/9045213/44a5dc57-eb28-497f-bb73-64e8de69f648)

When looking at a sandbox's folder unsandboxed like `C:\Sandbox\WDAGUtilityAccount\DefaultBox\user\current\Downloads`, why will `right-click > Open` cause the sandbox to run? 

Is there a way to make it open with Host's default program instead?
Like how it does with `right-click > Open with > program`

**Comment by edions** [comment]
When using "Open" within the sandbox folder, it defaults to running the program sandboxed to ensure that any potential changes or actions performed by the program do not affect the host system.

If you want an option to run it unsandboxed, you can enable it on:
`Global Settings > Shell Integration > Add 'Run Un-Sandboxed' to the context menu`

**Reply by TFWol** [reply]
But that doesn't address how the "Open with" is able to run unsandboxed

**Reply by edions** [reply]
No clue why opening it with the host's default program allows it to run unsandboxed. Maybe it's a bug or something, but placing the DefaultBox folder in the Forced Programs seems to fix the issue of it running unsandboxed.

Try putting this in Forced Programs.
`C:\Sandbox\WDAGUtiltyAccount\DefaultBox`

**Reply by hg421** [reply]
The reason why right click -> Open (or double clicking) starts the process in the sandbox is because in this case Windows explorer sets the folder containing the file as the working directory of the new process. But not when using "open with".

Since Sandboxie also looks at the initial working directory to decide whether to force a new process into the box or not, it leads to the observed behavior.

**Reply by isaak654** [reply, marked as answer]
> But that doesn't address how the "Open with" is able to run unsandboxed

I think this is part of the following issue: https://github.com/sandboxie-plus/Sandboxie/issues/1138

**Reply by TFWol** [reply]
> I think this is part of the following issue: #1138

Yeah, it seems like it. This part especially:

> Even if you open an instantce of explorer unsandboxed and go to a sandboxed file, for example let's say "c:\Sandbox\Usr\DefaultBox\D\file.docx" when you try to open the file normally it would open it sandboxed in the default app assigned to open that file type, but if you right click and choose "open with" and than another program that can open that file, that program will open unsandboxed even though that file is in a sandbox and should open sandboxed.

On my Main, I primarily run Security Hardened boxes with Data Protection so troubleshooting if it's a bug, feature, or setting I need to change can get exceedingly difficult.

**Comment by love-code-yeyixiao** [comment]
Sandman menu:"Sandbox"-> "Maintain"-> "Stop All"
And then you will lose security protection but you can feel free to run dangerous files in sandbox folders.

**Reply by TFWol** [reply]
...what?
